### PR TITLE
exclude deleted cases from query results

### PR DESCRIPTION
**Pull request by snopoke**
While working on the couch -> SQL migration I noticed that we don't exclude deleted cases from a bunch of queries.

@dimagi/scale-team 
buddy @biyeun 

best reviewed by commit :fish:

### CODE REVIEW

File: corehq/form_processor/tests/test_case_dbaccessor.py [hunk 1]
```diff
@@ -641,6 +641,28 @@ def test_closed_transactions_with_tracked(self):
...
+        self.assertEqual(case_ids, [case2.case_id])
+
+        deleted = CaseAccessorSQL.get_deleted_case_ids_in_domain(DOMAIN)
+        self.assertEquals(deleted, [case1.case_id])
+
+    def test_get_deleted_case_ids_by_owner(self):
+        user_id = uuid.uuid4().hex
+        case1 = _create_case(user_id=user_id)
+        case2 = _create_case(user_id=user_id)
+        case3 = _create_case(user_id=user_id)
```
Comment (dimagimon) [inline corehq/form_processor/tests/test_case_dbaccessor.py]: F841 local variable 'case3' is assigned to but never used

**Comment by snopoke** [comment]
Build failing due to ordering in sharded tests:
```
======================================================================
FAIL: test_get_deleted_case_ids_by_owner (corehq.form_processor.tests.test_case_dbaccessor.CaseAccessorTestsSQL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/form_processor/tests/test_case_dbaccessor.py", line 664, in test_get_deleted_case_ids_by_owner
    self.assertEqual(case_ids, [case1.case_id, case2.case_id])
AssertionError: Lists differ: [u'8c54715e5fce417d84131098a68... != [u'07198f67cb554b2d959c2328a92...
First differing element 0:
u'8c54715e5fce417d84131098a68968fc'
u'07198f67cb554b2d959c2328a926e7a6'
- [u'8c54715e5fce417d84131098a68968fc', u'07198f67cb554b2d959c2328a926e7a6']
+ [u'07198f67cb554b2d959c2328a926e7a6', u'8c54715e5fce417d84131098a68968fc']
```

**Comment by benrudolph** [comment]
good catch, looks good to me once test ✔️

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**